src/AutoEnvMap.php

```diff
@@ -100,4 +93,25 @@ private function paramToEnvName($parameterName, $envPrefix = '')
     {
         return $envPrefix . strtoupper(str_replace('.', '__', $parameterName));
     }
+
+    /**
+     * @param IncenteevFile $file
+     * @param string $param
+     * @param mixed $default
+     */
+    private function consumeParameter(IncenteevFile $file, $param, $default)
+    {
+        $fullParam = ($file->getAutoEnvFullname() ? $file->getParameterKey() . '.' : '') . $param;
+        $envName = $this->paramToEnvName($fullParam, $file->getAutoEnvPrefix());
+
+        if (getenv($envName) !== false) {
+            $this->envMap[$param] = $envName;
+        } elseif (is_array($default)) {
+            foreach ($default as $subParam => $subDefault) {
```

Comment:
Might `array_walk` be a more elegant solution here?

Reply:
Unfortunately not, an anonymous function would be required to prepend the parent parameter name  and those cannot call private methods in some versions of PHP.